diffprivlib/mechanisms/snapping.py

```diff
@@ -43,24 +45,19 @@ class Snapping(LaplaceTruncated):
     """
     def __init__(self, *, epsilon, sensitivity, lower, upper):
         super().__init__(epsilon=epsilon, sensitivity=sensitivity, delta=0.0, lower=lower, upper=upper)
-        self.epsilon = self._check_epsilon_machine_epsilon(epsilon)
         self._bound = self._scale_bound()
-        self._epsilon_0 = self.effective_epsilon()
-        self.scale = 1.0 / self._epsilon_0  # everything is scaled to sensitivity 1
-        self._lambda = self._get_nearest_power_of_2(self.scale)
 
-    @staticmethod
-    def _check_epsilon_machine_epsilon(epsilon):
-        machine_epsilon = np.finfo(float).epsneg
-        if epsilon < 2 * machine_epsilon:
+    @classmethod
+    def _check_epsilon_delta(cls, epsilon, delta):
+        if not (isinstance(epsilon, float) or isinstance(epsilon, np.float64)):
+            warnings.warn("The snapping mechanism expects epsilon to be a double precision floating-point number for"
+                          "precise rounding; epsilon will be cast to 64-bit float", DiffprivlibCompatibilityWarning)
```

Comment:
My apologies, I meant for the float check to be on the input value, not epsilon. This can be checked in `_check_all(value)`. Also, it may be best to do a quick sanity check like `float(value) != value`, since it should still be possible to input an integer, etc. Something like this:
```
def _check_all(self, value):
    super()._check_all(value)
    if float(value) != value:
        warnings.warn()

    return True
```

Reply:
Yes, this isn't quite ready.

The reason I'm checking `epsilon` is the two methods that depend on floating-point implementation details:`_get_nearest_power_of_2` and `effective_epsilon`, operate on `epsilon` (or values derived from it) not the input value. 

The other thing I was considering is to cast all values to np.longdouble, which is system dependant, and adapt the code to work with whatever precision that provides. It would complicate `_get_nearest_power_of_2` but would potentially lower the impact of the machine epsilon on the mechanism.